services/users.js

```diff
@@ -25,17 +26,16 @@ async function login (secret) {
 }
 
 // TODO: refactor this so it makes more sense
-async function addFriend (userId, friendFName, friendLName, friendEmail) {
+async function addFriend (userId, firstName, lastName, friendFName, friendLName, friendEmail) {
```

Comment:
Try to avoid adding arguments  to functions if you can help it. In this case, I think it's better to get user's firstName and lastName from the database.

Reply:
Since we already have the user data, shouldnt we limit the amount of operations we make to the db? Or does it not matter much?

Reply:
Gotta balance performance with maintainability. I'm not sure how long it takes, but it's probably quite fast. If this causes a bug or makes it annoying to change, then it's better to leave out a non-trivial optimization to make development easier.

We want to avoid the anti-pattern called [premature optimization](https://wiki.c2.com/?PrematureOptimization).

If we start running into issues with performance, then we should start with the exposure algorithm, which hits the DB for every single friend in the network (lol).

Reply:
Plus, the DB should be our Single Source of Truth for name info -- letting the request do it opens us up to bugs with inconsistent name info (or even attacks with malicious? name info)

Reply:
Ahh ok that makes sense. Thanks for clarifying.